__unported__/account_statement_completion_compassion/model/completion_rules.py

```diff
@@ -69,116 +118,102 @@ def get_from_partner_ref(self, cr, uid, id, st_line, context=None):
             'account_id' : value,
             ...}
         """
-        ref = st_line['ref'].strip()
+        ref = st_line.ref
         res = {}
-        partner_obj = self.pool.get('res.partner')
+        partner_obj = self.env['res.partner']
         partner_ids = partner_obj.search(
-            cr, uid,
-            [('ref', '=', str(int(ref[9:16]))), ('is_company', '=', False)],
-            context=context)
+            [('ref', '=', ref[9:16]),
+             ('is_company', '=', False)]).ids
```

Comment:
You could remove .ids (and rename variable to partners) in order to avoid the browse later.